### PR TITLE
Fix Loader type in webpack

### PR DESCRIPTION
**Pull request by jason0x43**
The sourceMap parameter passed to loaders, and that loaders pass to the
callback, is a `RawSourceMap`, not a `string | Buffer`. See the [webpack loader docs](https://webpack.js.org/api/loaders/).

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/webpack/index.d.ts [hunk 1]
```diff
@@ -9,12 +9,15 @@
 //                 Ahmed T. Ali <https://github.com/ahmed-taj>
 //                 Alan Agius <https://github.com/alan-agius4>
 //                 Spencer Elliott <https://github.com/elliottsj>
+//                 Jason Cheatham <https://github.com/jason0x43>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
```
Comment (alan-agius4) [inline types/webpack/index.d.ts]: Why is this necessary now to build using TS 2.2 instead of 2.7?

**Review by alan-agius4 — CHANGES_REQUESTED** [review]

File: types/webpack/index.d.ts (hunk 1 above)
Comment (jason0x43) [reply types/webpack/index.d.ts]: I added a version line (there wasn't one to begin with) because the default version used in CI is 2.1, which doesn't understand `object` (which I was using in the self-tests). However, specifying 2.2 here breaks other packages that depend on webpack, like case-sensitive-paths-webpack-plugin, because 2.2 is more recent than the default 2.1. Using anything higher than 2.2 fails local linting. (It may still pass CI, but then what's the point of the local linter?)

My latest version just removes the version line again and uses the specific `RawSourceMap` type in the tests rather than generic `object`.

**Review by alan-agius4 — APPROVED** [review]

File: types/webpack/package.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "dependencies": {
```
Comment (alan-agius4) [inline types/webpack/package.json]: Set this to private.

**Review by alan-agius4 — CHANGES_REQUESTED** [review]

File: types/webpack/package.json (hunk 2 above)
Comment (jason0x43) [reply types/webpack/package.json]: _sigh_ That's another thing that causes the local linter to fail but apparently is required in CI. Either I'm using the local linter wrong (just `npm run lint webpack`, right?), or the README needs to be updated.

**Comment by alan-agius4** [comment]
Yes, thats the correct command.

File: types/webpack/package.json (hunk 2 above)
Comment (alan-agius4) [reply types/webpack/package.json]: Yes, thats the correct command.

**Comment by jason0x43** [comment]
Well then, I guess those tslint rules were necessary.

Is there some way to locally run the same checks that CI runs?

**Comment by alan-agius4** [comment]
You can try ‘npm test’

**Review by alan-agius4 — APPROVED** [review]

**Comment by jason0x43** [comment]
So, it turns out my problem was that I was using a very old version of types-publisher. The DT package.json uses a git ref (`types-publisher#production`) rather than a versioned package, and my local package-lock.json file was locked to a commit of types-publisher from about 10 months ago. Since the version of TP never changed in the DT package.json, I never got updates when running `npm install`.

**Comment by jason0x43** [comment]
Lesson of the day: remove `package-lock.json` and `node_modules` when updating DT.

**Comment by DanielRosenwasser** [comment]
Thanks!
